src/map/Map.js

```diff
@@ -696,7 +696,8 @@ L.Map = L.Evented.extend({
 		for (var i = 0; i < targets.length; i++) {
 			if (targets[i].listens(type, true)) {
 				targets[i].fire(type, data, true);
-				if (data.originalEvent._stopped) { return; }
+				if (data.originalEvent._stopped
+					|| (targets[i].options.unpropagateEvents && targets[i].options.unpropagateEvents.indexOf(type) !== -1)) { return; }

```

Comment:
Huuuu, good catch! Should we switch to using an array instead? Not sure about perfs :/